src/index.js

```diff
@@ -74,9 +75,15 @@ function serve (options = { contentBase: '' }) {
 
   // If HTTPS options are available, create an HTTPS server
   if (options.https) {
-    server = createHttpsServer(options.https, requestListener).listen(options.port, options.host)
+    server = createHttpsServer(options.https, requestListener)
+    server.listen(options.port, options.host, () => {
+      options.onListening(server)
+    })
   } else {
-    server = createServer(requestListener).listen(options.port, options.host)
+    server = createServer(requestListener)
+    server.listen(options.port, options.host, () => {
+      options.onListening(server)
+    })
```

Comment:
How about reusing the listen params + callback
```suggestion
  server = options.https ? createHttpsServer(options.https, requestListener) : createServer(requestListener)
  server.listen(options.port, options.host, () => {
    options.onListening(server)
  })
```